### PR TITLE
Promote CSI raw blocks to beta

### PR DESCRIPTION
**Pull request by bswartz**
**What type of PR is this?**
> /kind feature

**What this PR does / why we need it**:
Promote the CSIBlockVolume feature to beta.

**Does this PR introduce a user-facing change?**:
```release-note
The CSIBlockVolume feature gate is now beta, and defaults to enabled.
```

### CODE REVIEW

**Comment by msau42** [comment]
As part of this PR, can you:
* temporarily update the csi hostpath driver version to canary (now that https://github.com/kubernetes-csi/csi-driver-host-path/pull/6 has merged)
* Toggle CapBlock to true for csi hostpath
* Let's see if the block tests get run in pull-kubernetes-e2e-gce

**Comment by msau42** [comment]
Here's the hostpath driver configs: 
https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/test/e2e/storage/drivers/csi.go#L86
https://sourcegraph.com/github.com/kubernetes/kubernetes/-/tree/test/e2e/testing-manifests/storage-csi/hostpath/hostpath

**Comment by wongma7** [comment]
pod events:

I0305 04:53:59.684] Mar 5 04:53:31.152: INFO: At 2019-03-05 04:49:09 +0000 UTC - event for security-context-e17ae819-3f01-11e9-a91a-0e847381c6db: {attachdetach-controller } SuccessfulAttachVolume: AttachVolume.Attach succeeded for volume "pvc-db295e29-3f01-11e9-acd4-42010a280002"
66516
I0305 04:53:59.684] Mar 5 04:53:31.152: INFO: At 2019-03-05 04:49:19 +0000 UTC - event for security-context-e17ae819-3f01-11e9-a91a-0e847381c6db: {kubelet e2e-71257-674b9-minion-group-g244} FailedMapVolume: MapVolume.SetUp failed for volume "pvc-db295e29-3f01-11e9-acd4-42010a280002" : rpc error: code = Internal desc = failed to create target path: /var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-db295e29-3f01-11e9-acd4-42010a280002: open /var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-db295e29-3f01-11e9-acd4-42010a280002: no such file or directory
66517
I0305 04:53:59.685] Mar 5 04:53:31.152: INFO: At 2019-03-05 04:50:15 +0000 UTC - event for security-context-e17ae819-3f01-11e9-a91a-0e847381c6db: {kubelet e2e-71257-674b9-minion-group-g244} FailedMount: Unable to mount volumes for pod "security-context-e17ae819-3f01-11e9-a91a-0e847381c6db_volumemode-5198(e17bcbb0-3f01-11e9-acd4-42010a280002)": timeout expired waiting for volumes to attach or mount for pod "volumemode-5198"/"security-context-e17ae819-3f01-11e9-a91a-0e847381c6db". list of unmounted volumes=[volume1]. list of unattached volumes=[volume1 default-token-rk7j6]


kubelet logs:

I0305 04:49:19.839965    1620 csi_block.go:179] kubernetes.io/csi: blockMapper.publishVolumeForBlock created directory for publishPath successfully [/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish]
E0305 04:49:19.881850    1620 csi_block.go:201] kubernetes.io/csi: blockMapper.publishVolumeForBlock failed: rpc error: code = Internal desc = failed to create target path: /var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-db295e29-3f01-11e9-acd4-42010a280002: open /var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-db295e29-3f01-11e9-acd4-42010a280002: no such file or directory

Where hostpath csi driver returns that error:
https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/pkg/hostpath/nodeserver.go#L87

File: test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ spec:
           - mountPath: /registration
             name: registration-dir
         - name: hostpath
-          image: quay.io/k8scsi/hostpathplugin:v1.0.1
+          image: quay.io/k8scsi/hostpathplugin:canary
```
Comment (wongma7) [inline test/e2e/testing-manifests/storage-csi/hostpath/hostpath/csi-hostpathplugin.yaml]: need the latest changes from https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/hostpath/csi-hostpath-plugin.yaml
e.g.
```
        - hostPath:
            path: /var/lib/kubelet/plugins
            type: Directory
            name: plugins-dir
```

**Comment by msau42** [comment]
Failing tests are not related to block.

I went through passed tests and I found one test case for raw block with CSI:
```
[sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (block volmode)] volumeMode should create sc, pod, pv, and pvc, read/write to the pv, and delete all created resources
```

@mkimuram @wongma7  are there more block tests that are supposed to run?

I think this is supposed to run but didn't?
* provisioning.go: should create and delete block persistent volumes

**Comment by msau42** [comment]
There's also this failure test case for drivers that dont support block:
```
[sig-storage] CSI Volumes [Driver: csi-hostpath-v0] [Testpattern: Dynamic PV (block volmode)] volumeMode should fail in binding dynamic provisioned PV to PVC
```

**Comment by wongma7** [comment]
those 2 tests ran in the latest results:
[sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (default fs)] provisioning should create and delete block persistent volumes 	55s
[sig-storage] CSI Volumes [Driver: csi-hostpath-v0] [Testpattern: Dynamic PV (block volmode)] volumeMode should fail in binding dynamic provisioned PV to PVC 	5m19s
there aren't any other tests specifically for block, we test provisioning, r/w, & binding. Other test suites: `volumeio` writes to files & `volumes` tests different filesystems etc.

**Comment by mkimuram** [comment]
> there aren't any other tests specifically for block

Right. And once https://github.com/kubernetes/kubernetes/pull/74693 is merged, these tests will be also tested, except the tests that require different nodes. (Because hostpath driver will require specific node.)

**Comment by msau42** [comment]
Maybe @mkimuram can you include these commits in your PR so that we can immediately validate the new tests against the csi hostpath driver?

**Comment by msau42** [comment]
/lgtm
The same in-tree raw block tests are testing the hostpath csi driver now and are passing.

More test coverage in general for in-tree + csi will be part of https://github.com/kubernetes/kubernetes/pull/74693

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Pull request merged**